### PR TITLE
[qtmozembed] Automated tests fix. Fixes JB#56513 OMP#JOLLA-558

### PR DESCRIPTION
**Pull request by mkenttala**

### CODE REVIEW

File: tests/auto/desktop-qt5/newviewrequest/tst_newviewrequest.qml [hunk 1]
```diff
@@ -63,14 +69,14 @@ Item {
             verify(MyScript.waitLoadFinished(mozView))
             compare(mozView.title, "NewWinExample")
             verify(MyScript.wrtWait(function() { return (!mozView.painted); }))
+            mozView.loadFrameScript("chrome://tests/content/testHelper.js")
```
Comment (rainemak) [inline tests/auto/desktop-qt5/newviewrequest/tst_newviewrequest.qml:72]: This is too late...Could go to ViewComponents.qml:

```
Component.onCompleted: loadFrameScript("chrome://tests/content/testHelper.js")
```

This way all tests that are using dynamically allocated WebView would get it. mozview_p itself takes care of loading it after the view is initialized.

File: tests/auto/desktop-qt5/passwordmgr/tst_passwordmgr.qml [hunk 2]
```diff
@@ -47,6 +47,7 @@ Item {
         name: "tst_passwordmgr"
         when: windowShown
 
+        /* Not working with esr78
```
Comment (rainemak) [inline tests/auto/desktop-qt5/passwordmgr/tst_passwordmgr.qml:50]: Please create a task out of this.

File: tests/auto/desktop-qt5/useragent/tst_useragent.qml [hunk 3]
```diff
@@ -12,12 +13,34 @@ Item {
...
         visible: true
         focus: true
         active: true
         anchors.fill: parent
+        property string useragent: ""
+
+        onViewInitialized: {
+            webViewport.loadFrameScript("chrome://tests/content/testHelper.js")
+            webViewport.addMessageListener("testembed:useragent")
+        }
```
Comment (rainemak) [inline tests/auto/desktop-qt5/useragent/tst_useragent.qml:34]: No need to handle these on viewInitialized signal handler rather Component.onCompleted is enough.

File: tests/auto/desktop-qt5/useragent/tst_useragent.qml [hunk 4]
```diff
@@ -59,12 +82,15 @@ Item {
...
+            webViewport.useragent = ""
+            verify(webViewport.useragent === "")
             loadPage(test_page)
-            verify(webViewport.httpUserAgent === override_default)
+            webViewport.sendAsyncMessage("embedtest:useragent", {})
+            verify(MyScript.wrtWait(function() { return (webViewport.useragent === ""); }))
+            verify(webViewport.useragent === override_default)
         }
 
+        /* QuickMozView::httpUserAgent is always empty with current esr78 engine
```
Comment (rainemak) [inline tests/auto/desktop-qt5/useragent/tst_useragent.qml:93]: Please create task and link bz / jira link to this comment (JB#xyz / JIRA-xyz).

File: tests/auto/desktop-qt5/downloadmgr/tst_downloadmgr.qml [hunk 5]
```diff
@@ -40,7 +40,7 @@ Item {
         focus: true
         active: true
         anchors.fill: parent
-        onViewInitialized: {
+        Component.onCompleted: {
             webViewport.addMessageListener("embed:downloadpicker")
             appWindow.mozViewInitialized = true
```
Comment (rainemak) [inline tests/auto/desktop-qt5/downloadmgr/tst_downloadmgr.qml:45]: addMessageListener, loadFrameScript, and e.g. setPreference can be called before view initialized and internals take care of handling them. That said, "appWindow.mozViewInitialized = true" should be onViewInitialized.

**Review by rainemak — APPROVED** [review]
I'm not giving completely objective review as I also contributed to this PR. That said, now testrunner-lite reports constantly (at least for me thus far) "Executed 16 cases. Passed 16 Failed 0"

**Review by okodron — APPROVED** [review]

**Pull request merged**